### PR TITLE
Update the Eclipse code of conduct to version 2

### PR DESCRIPTION
**Pull request by pshipton**
A copy of https://raw.githubusercontent.com/eclipse/.github/master/CODE_OF_CONDUCT.md

See the web version https://www.eclipse.org/org/documents/Community_Code_of_Conduct.php

Similar to [the PR](https://github.com/eclipse-openj9/openj9/pull/13637) for the original code of conduct , we're asking all committers, and contributors, to approve, indicating they have read the updated code of conduct and approve the change. We're not looking for feedback on the wording, which is copied from the Eclipse Foundation.

### CODE REVIEW

**Comment by DanHeidinga** [comment]
@pshipton do you have a description of what changed between this and the previous version?  The diff is not great for seeing the actual changed parts

**Comment by pshipton** [comment]
> do you have a description of what changed

I do not.

**Comment by pshipton** [comment]
I'll start with a review from the project leads, and Keith (for nitpicks). Afterwards I'll make a request on Slack to all the other committers and official contributors, https://projects.eclipse.org/projects/technology.openj9/who

**Review by keithc-ca — APPROVED** [review]
Identical to the file at https://github.com/eclipse/.github (with trailing whitespace removed).

**Comment by keithc-ca** [comment]
Perhaps we should follow the practice of the Adoptium project which incorporates the code of conduct by reference and then, as they did, delete it from our repositories. See https://github.com/adoptium and, for example, https://github.com/adoptium/infrastructure/pull/2870.

**Comment by pshipton** [comment]
I need to check the latest release review requirements. The file used to be required, although perhaps it can just contain a link.

**Comment by pshipton** [comment]
Looking a recent release [review template](https://gitlab.eclipse.org/eclipsefdn/emo-team/emo/-/issues/443), it still lists CODE_OF_CONDUCT as a recommended file, using https://github.com/eclipse/.github/blob/master/CODE_OF_CONDUCT.md as an example. https://github.com/adoptium/infrastructure/pull/2870 states "No longer needed as it's pulled from our central repo", which implies it's still around, copied to the right places. Looking at https://github.com/adoptium/adoptium-support there is a side bar on the right, which contains https://github.com/adoptium/.github/blob/main/CODE_OF_CONDUCT.md which contains the text and not a link to the Eclipse site.

**Comment by pshipton** [comment]
Looking at https://github.com/eclipse-openj9/openj9, there is also a side bar on the right which contains a link to https://github.com/eclipse-openj9/openj9/blob/master/CODE_OF_CONDUCT.md

**Comment by DanHeidinga** [comment]
This is a pretty major change to the Code of Conduct and after my quick reading of the history, changes the tone on enforcement and grants more control to the Foundation and less to the Project for addressing issues.

I think we probably need a committer vote on updating? Thoughts @pshipton @mstoodle?

**Comment by pshipton** [comment]
I tweaked the wording in the description to include approval of the change.

**Comment by mstoodle** [comment]
I support having a committer vote to acknowledge everyone has read and understands the updated code of conduct. The changes are interesting. A suspicious person would wonder who the heck is this "Eclipse Foundation Conduct Committee" and how is its membership determined / changed. Given its nature, it won't be able to operate completely transparently and one can only hope that it doesn't become a control point in the Eclipse Foundation.

I vote to not reject the code of conduct and commit to (continue to) behave according to its spirit.

**Review by ChengJin01 — APPROVED** [review]

**Review by ChengJin01 — APPROVED** [review]

**Review by gacholio — APPROVED** [review]

**Comment by pshipton** [comment]
We're limited to request review from 15 reviewers. I've post the review request in Slack, and I'll track down anyone left over.

**Review by fengxue-IS — APPROVED** [review]

**Comment by jdmpapin** [comment]
> The diff is not great for seeing the actual changed parts

Just next to the "viewed" check box for the file, there's a pair of buttons you can use to select between "display the source diff" and "display the rich diff":

image

The rich diff seems to show the differences much more clearly.

**Review by dmitripivkine — APPROVED** [review]

**Review by mpirvu — APPROVED** [review]

**Review by ymanton — APPROVED** [review]

**Review by babsingh — APPROVED** [review]

**Review by JasonFengJ9 — APPROVED** [review]

**Review by AdamBrousseau — APPROVED** [review]

**Review by hzongaro — APPROVED** [review]

**Review by amicic — APPROVED** [review]

**Review by dsouzai — APPROVED** [review]

**Review by 0xdaryl — APPROVED** [review]

**Review by jdmpapin — APPROVED** [review]

**Review by hangshao0 — APPROVED** [review]

**Review by smlambert — APPROVED** [review]

**Review by mstoodle — APPROVED** [review]

**Review by knn-k — APPROVED** [review]

**Review by doveye — APPROVED** [review]

**Comment by pshipton** [comment]
@Sreekala-Gopakumar can you pls review this as well.

**Review by tajila — APPROVED** [review]

**Review by zl-wang — APPROVED** [review]

**Review by nbhuiyan — APPROVED** [review]

**Review by jdekonin — APPROVED** [review]

**Review by vijaysun-omr — APPROVED** [review]

**Review by joransiu — APPROVED** [review]

**Review by llxia — APPROVED** [review]

**Comment by pshipton** [comment]
@DanHeidinga I believe we have everyone, this can be merged with you approval.

**Review by DanHeidinga — APPROVED** [review]

**Comment by Sreekala-Gopakumar** [comment]
> @Sreekala-Gopakumar can you pls review this as well.

Read and approved.

**Pull request merged**
